src/linesearches_compat.jl

```diff
@@ -0,0 +1,234 @@
+
+"""
+    LineSearchesStepsize <: Stepsize
+
+Wrapper for line searches available in the `LineSearches.jl` library.
+
+## Constructors
+
+    LineSearchesStepsize(
+        M::AbstractManifold,
+        ls,
+        is;
+        retraction_method::AbstractRetractionMethod=default_retraction_method(M),
+        vector_transport_method::AbstractVectorTransportMethod=default_vector_transport_method(M),
+    )
+    LineSearchesStepsize(
+        ls,
```

Comment:
I would prefer speaking names and if possible types (even if its the `Optim.` ones.
This is `line_search`? the next is `initial_step`?

Reply:
Yes, I will expand these names.